plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -52,6 +52,25 @@ export function castingCompare(
     b: any,
     operator: PropertyOperator.Exact | PropertyOperator.IsNot | PropertyOperator.LessThan | PropertyOperator.GreaterThan
 ): boolean {
+    // Check null case first
+    if (a === null || b === null) {
```

Comment:
I suppose this should be in the `if (typeof a !== typeof b)` block, since this `"null"` vs. `null` comparison only is relevant when one of the values is a string, and the other is a null.

Reply:
That doesn't work, since the `a != b` in the basic check short circuits first, as `"null" != null` 😬

Reply:
Ohh, right, so maybe we should do `if (a === null) a = "null"; if (b === null) b = "null"`? Trying to somehow reduce the line count here 😅